projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -1,108 +1,191 @@
+    this.messageService.add({
+      severity: "error",
+      summary: "Error",
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
+    });
+  }
+
+  isAccepting() {
+    return this.onGoingActivity === Activity.Accepting;
+  }
+
+  isRejecting() {
+    return this.onGoingActivity === Activity.Rejecting;
+  }
+
+  isIdle() {
+    return this.onGoingActivity === Activity.None;
+  }
+
+  constructor(
+    private groupService: GroupService,
+    private messageService: MessageService
+  ) {}
 
   ngOnInit() {
     this.panel.push({
       name: "Pending Requests",
       columns: this.columns,
     });
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
   }
 
-  ngOnChanges(changes: SimpleChanges) {
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
+  ngOnChanges(_changes: SimpleChanges) {
+    this.selection = [];
+    this._reloadData();
   }
 
-  onExpandWidth(e) {}
+  onAcceptOrReject(action: Action) {
+    if (this.selection.length === 0 || this.onGoingActivity !== Activity.None) {
+      return;
+    }
+
+    let resultObserver: Observable<RequestActionResponse>;
+    this.onGoingActivity = (action === Action.Accept) ? Activity.Accepting : Activity.Rejecting;
+    
+    const group_ids = this.selection.map((req: PendingRequest) => req.joining_user.group_id);
+
+    if (action === Action.Accept) {
+      resultObserver = this.groupService.acceptJoinRequest(this.id, group_ids);
+    } else {
+      resultObserver = this.groupService.rejectJoinRequest(this.id, group_ids);
+    }
 
-  onClickAccept(e) {
-    this.onAccept.emit(e);
+    resultObserver.subscribe(
+      (res: RequestActionResponse) => {
+        this._displayResponseToast(
+          res,
+          action,
+          action === Action.Accept ? "accepted" : "declined"
+        );
+        if (res.success === true && res.message === "updated") {
+          this._reloadData();
+        }
+        this.onGoingActivity = Activity.None;
+        this.selection = [];
+      },
+      (err) => {
+        this._processRequestError(err);
+        this.onGoingActivity = Activity.None;
+      }
+    );
   }
 
-  onClickReject(e) {
-    this.onReject.emit(e);
+  onSelectAll(_event) {
+    if (this.selection.length === this.requests.length) {
+      this.selection = [];
+    } else {
+      this.selection = this.requests;
+    }
   }
 
   onCustomSort(event: SortEvent) {
-    let diff = false;
-
-    const sortBy = event.multiSortMeta.map((meta) =>
+    const sortMeta = event.multiSortMeta.map((meta) =>
       meta.order === -1 ? `-${meta.field}` : meta.field
     );
 
-    if (sortBy.sort().join(" ") !== this.prevSortMeta) {
-      diff = true;
-    }
-
-    if (!diff) {
-      return;
+    if (!_.isEqual(_.sortBy(sortMeta), _.sortBy(this.currentSort))) {
```

Comment:
You compare two sorting method by comparing them with their parameters sorted? So you consider [id, at] being the same sort as [at, id]? 🤔

Reply:
That's really wrong if we consider order.
Anyway, the `event.multiSortMeta` doesn't recognize the clicked field order and its response has the same order (A->Z) according to its field name.
For the pending requests, its response order is the same.
![image](https://user-images.githubusercontent.com/51055029/82542283-4c851580-9b84-11ea-9526-6133115cef58.png)

Reply:
I don't get what you want to show me. Actually, the table is not supported multisort anymore currently, wasn't it earlier?

Reply:
I mean, PrimeNG table doesn't remember the order of fields to be sorted.
Let's assume that I click on the `member_id` first and then click on `at`.
Then the multiSortMeta should be ordered like `member_id`, `at`.
But it sorts the array automatically by the length of its field name so the multiSortMeta is ordered like `at`, `member_id`.
That's the point what I am going to explain.

Reply:
If I sort by `id` and then `at`, I have it in the right order. If I do the opposite, I have the opposite order. 
![Screenshot 2020-05-21 at 17 36 05](https://user-images.githubusercontent.com/1053150/82576079-a2a88780-9b89-11ea-9e18-57eac5121c72.png)

Reply:
Yes, it will.
I removed the multiSortMeta after you pointed it out.
It started working then, the reason I wasn't getting the right order was that I didn't reset the sort order in the UI.
No problem, for now, I removed the sortBy function in comparing the sortMeta.
fixed.